_source/_posts/2020-04-29-master-filestream-in-csharp-aspnetcore.md

```diff
@@ -0,0 +1,321 @@
+---
+layout: blog_post
+title: "How to Master the Filestream in C#"
+author: chase-aucoin
+by: contractor
+communities: [.net]
+description: "Hesitant about when to use the Filestream in C#? This is a primer that explains how to understand the concept and securely stream files hosted on AWS."
+tags: [filestream, C#, csharp, dotnet, .net, asp.net]
+tweets:
+- "When should a dev use the #filestream in #csharp? We explain the concept and how to securely stream files hosted in #aw ->."
+- "Unsure about using the #Filestream in #dotnet? We've got a primer just waiting for you ->"
+- "Learn how to create your own secure streaming service in #dotnetcore with the #filestream and #aws ->"
+image: blog/featured/okta-dotnet-half.jpg
+type: conversion
+---
+
+We live in a world that moves **fast**. Compared to the mid 90s through early 2000s, we have incredibly intelligent  technology. Effectively, we have super-computers in our pockets. Our *actual*, modern supercomputers would have seemed like works of fiction just two decades ago. Not only is our ability to compute fast, but so is our data - with cellular 4G averaging 18.1 Mbps and 5G coming in at an average of 111.8 Mbps, at the time of writing this.
+
+{% img blog/filestream-csharp/01-kbsize.png alt:"Data speeds" width:"800" %}{: .center-image }
+
+With all this speed, there has been abstraction after abstraction placed on top of our data and connections to make development easier, but there is a cost to that ease. We send *a lot* of data over the wire. For all of the assets on a webpage, websites targeting desktops send almost 2.1Mb of data, while mobile sends nearly 1.9Mb. These speeds were not always this ubiquitous. Perhaps it's time to look back at how applications were able to provide relatively fast interaction with data in a world long forgotten: The world of 56Kbps.
+
+My goal is that, by the end of this article, you have a stronger understanding of some of the low level APIs that make moving large amounts of data possible. Additionally, we're going to use this knowledge to make a secure streaming service from Amazon Web Services S3 that will let us restrict data to certain roles. We'll do this using the smallest instance available on AWS,  demonstrating the power of streams to move big data with small machines.
+
+## Prerequisites
```

Comment:
I would make this bold rather than a heading since H2's are very important for SEO and this doesn't have any keywords in it.

Reply:
Good idea. Changed :)